src/components/Dropdown/Dropdown.tsx

```diff
@@ -421,50 +424,43 @@ export default class Dropdown extends AutoControlledComponent<
         }}
       >
         <List
-          {...accessibilityMenuPropsRest}
+          {...accessibilityMenuProps}
           styles={styles.list}
           tabIndex={search ? undefined : -1} // needs to be focused when trigger button is activated.
           aria-hidden={!isOpen}
-          items={isOpen ? this.renderItems(styles, variables, getItemProps, highlightedIndex) : []}
+          items={isOpen ? this.renderItems(variables, getItemProps, highlightedIndex) : []}
         />
       </Ref>
     )
   }
 
   private renderItems(
-    styles: ComponentSlotStylesInput,
     variables: ComponentVariablesInput,
     getItemProps: (options: GetItemPropsOptions<ShorthandValue>) => any,
     highlightedIndex: number,
   ) {
-    const { renderItem, noResultsMessage } = this.props
+    const { loading, loadingMessage, noResultsMessage, renderItem } = this.props
+
     const filteredItems = this.getItemsFilteredBySearchQuery()
+    const items = filteredItems.map((item, index) =>
+      DropdownItem.create(item, {
+        defaultProps: {
+          active: highlightedIndex === index,
+          variables,
+          ...(typeof item === 'object' &&
+            !item.hasOwnProperty('key') && {
+              key: (item as any).header,
+            }),
+        },
+        overrideProps: () => this.handleItemOverrides(item, index, getItemProps),
+        render: renderItem,
+      }),
+    )
 
-    if (filteredItems.length > 0) {
-      return filteredItems.map((item, index) => {
-        return DropdownItem.create(item, {
-          defaultProps: {
-            active: highlightedIndex === index,
-            variables,
-            ...(typeof item === 'object' &&
-              !item.hasOwnProperty('key') && {
-                key: (item as any).header,
-              }),
-          },
-          overrideProps: () => this.handleItemOverrides(item, index, getItemProps),
-          render: renderItem,
-        })
-      })
-    }
-    // render no match message.
     return [
-      noResultsMessage
-        ? {
-            key: 'dropdown-no-results',
-            content: <Text weight="bold" content={noResultsMessage} />,
-            styles: styles.emptyListItem,
-          }
-        : null,
+      ...items,
+      loading && DropdownMessageLoading.create(loadingMessage),
```

Comment:
this `DropdownMessageLoading ` component seems to be extremely simple in terms of its rendering logic - couldn't we reuse `Box` component here instead of creating new one?

Reply:
I reused `ListItem` there